### PR TITLE
docs: update cms-sanity example

### PR DESCRIPTION
**Pull request by stipsan**
Upgrades `next-sanity` to the latest version and applies breaking changes.
Improves the docs for the Sanity example.

## Documentation / Examples

- [x] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

File: examples/cms-sanity/README.md [hunk 1]
```diff
@@ -168,3 +194,8 @@ To deploy your local project to Vercel, push it to GitHub/GitLab/Bitbucket and [
 Alternatively, you can deploy using our template by clicking on the Deploy button below.
 
 [![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https://github.com/vercel/next.js/tree/canary/examples/cms-sanity&project-name=cms-sanity&repository-name=cms-sanity&env=NEXT_PUBLIC_SANITY_PROJECT_ID,SANITY_API_TOKEN,SANITY_PREVIEW_SECRET&envDescription=Required%20to%20connect%20the%20app%20with%20Sanity&envLink=https://vercel.link/cms-sanity-env)
+
+#### Next steps
+
+- Invalidate your routes in production on-demand with GROQ powered webhooks
```
Comment (leerob) [inline examples/cms-sanity/README.md:200]:
```suggestion
- Invalidate your routes in production [on-demand](https://nextjs.org/blog/next-12-1#on-demand-incremental-static-regeneration-beta) with GROQ powered webhooks
```

File: examples/cms-sanity/components/more-stories.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ export default function MoreStories({ posts }) {
       <h2 className="mb-8 text-6xl md:text-7xl font-bold tracking-tighter leading-tight">
         More Stories
       </h2>
-      <div className="grid grid-cols-1 md:grid-cols-2 md:col-gap-16 lg:col-gap-32 row-gap-20 md:row-gap-32 mb-32">
+      <div className="grid grid-cols-1 md:grid-cols-2 md:gap-x-16 lg:gap-x-32 gap-y-20 md:gap-y-32 mb-32">
```
Comment (leerob) [inline examples/cms-sanity/components/more-stories.js:9]: This should be fixed in canary now 😄

File: examples/cms-sanity/components/post-body.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
 import markdownStyles from './markdown-styles.module.css'
-import BlockContent from '@sanity/block-content-to-react'
+import { PortableText } from '@portabletext/react'
```
Comment (leerob) [inline examples/cms-sanity/components/post-body.js:2]: Why was this changed?

File: examples/cms-sanity/lib/sanity.js [hunk 4]
```diff
@@ -1,7 +1,5 @@
-import {
-  createImageUrlBuilder,
-  createPreviewSubscriptionHook,
-} from 'next-sanity'
+import createImageUrlBuilder from '@sanity/image-url'
```
Comment (leerob) [inline examples/cms-sanity/lib/sanity.js:1]: Is there a reason this was removed from the next-sanity package?

Comment (balazsorban44) [reply examples/cms-sanity/lib/sanity.js:1]: Similar as https://github.com/vercel/next.js/pull/34898#discussion_r816949741 I assume:

https://github.com/sanity-io/next-sanity#createimageurlbuilder-is-removed

Comment (stipsan) [reply examples/cms-sanity/lib/sanity.js:1]: Yes, we're working on a better way to handle images that isn't using the image url builder pattern.
And we won't use this image builder utility.

**Review by leerob — APPROVED** [review]
